tests/functional/oauth_settings_clients.js

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern',
+  'intern!object',
+  'tests/lib/helpers',
+  'tests/functional/lib/helpers'
+], function (intern, registerSuite, TestHelpers, FunctionalHelpers) {
+  const config = intern.config;
+  const CONTENT_SERVER = config.fxaContentRoot;
+  const APPS_SETTINGS_URL = CONTENT_SERVER + 'settings/clients?forceDeviceList=1&forceAppsList=1';
+  const UNTRUSTED_OAUTH_APP = config.fxaUntrustedOauthApp;
+
+  const PASSWORD = 'password';
+  const UNTRUSTED_OAUTH_WINDOW = 'untrusted';
+
+  const thenify = FunctionalHelpers.thenify;
+  const click = FunctionalHelpers.click;
+  const clearBrowserState = thenify(FunctionalHelpers.clearBrowserState);
+  const fillOutSignUp = thenify(FunctionalHelpers.fillOutSignUp);
+  const getVerificationLink = thenify(FunctionalHelpers.getVerificationLink);
+  const testElementExists = FunctionalHelpers.testElementExists;
+  const openFxaFromRp = FunctionalHelpers.openFxaFromRp;
+  const openPage = FunctionalHelpers.openPage;
+  const pollUntilGoneByQSA = FunctionalHelpers.pollUntilGoneByQSA;
+  const type = FunctionalHelpers.type;
+  const closeCurrentWindow = FunctionalHelpers.closeCurrentWindow;
+
+  var email;
+
+  registerSuite({
+    name: 'oauth settings clients',
+
+    beforeEach: function () {
+      email = TestHelpers.createEmail();
+
+      return this.remote
+        .then(clearBrowserState(this, {
+          '123done': true,
+          contentServer: true
+        }));
+    },
+
+    'rp listed in apps, can be deleted': function () {
+      this.timeout = 90 * 1000;
+
+      return openFxaFromRp(this, 'signup')
+        .then(fillOutSignUp(this, email, PASSWORD))
+        .then(testElementExists('#fxa-confirm-header'))
+        .then(getVerificationLink(email, 0))
+        .then((verificationLink) => {
```

Comment:
Are we allowed to use fat arrows in our functional tests before we are fully ES6 ready?